packages/untp-test-suite/integration/cli/untpTest.integration.test.ts

```diff
@@ -0,0 +1,228 @@
+      });
+    });
+  });
+
+  describe('process test runner and final report return FAIL', () => {
+    let credentialFileName: string;
+    let storePath: string;
+    let stdout: any;
+    let credentials: ConfigContent[];
+
+    const mockPath = `${process.cwd()}/integration/mock/untpTestFail`;
+    beforeAll((done) => {
+      credentialFileName = 'credentialsExample.json';
+      storePath = `${mockPath}/${credentialFileName}`;
+      const fileContent = fs.readFileSync(storePath, 'utf8');
+
+      try {
+        credentials = JSON.parse(fileContent).credentials;
+      } catch (error) {
+        expect(error).toBeNull();
+      }
+
+      exec(`yarn untp test -c ${storePath}`, (error, result) => {
+        if (error) {
+          console.error(`execSync error: ${error}`);
+          return;
+        }
+
+        stdout = result;
+        done();
+      });
+    });
+
+    it('should show the certification in objectEvent data is object ', () => {
+      // get only data path of object event
+      const data = fs.readFileSync(`${process.cwd()}/${credentials[1].dataPath}`, 'utf8');
+      const jsonData = JSON.parse(data);
+      expect(jsonData.certification).toEqual(expect.any(Object));
+    });
+
+    describe('generate test suite result by template', () => {
+      it('should show FAIL in the report when data invalidate', () => {
+        expect(stdout).toMatch(/FAIL/);
+        expect(stdout).toMatch(/Your credentials are not UNTP compliant/);
+        expect(stdout).toContain('certification field must be array');
+        expect(stdout).toContain('type field should have required property');
+        expect(stdout).toContain('version field should have required property');
+        expect(stdout).toContain('dataPath field should have required property');
+      });
+    });
+  });
+
+  describe('process test runner and final report return WARN', () => {
+    let credentialFileName: string;
+    let storePath: string;
+    let stdout: any;
+    let credentials: ConfigContent[];
+
+    const mockPath = `${process.cwd()}/integration/mock/untpTestWarn`;
+    beforeAll((done) => {
+      credentialFileName = 'credentialsExample.json';
+      storePath = `${mockPath}/${credentialFileName}`;
+      const fileContent = fs.readFileSync(storePath, 'utf8');
+
+      try {
+        credentials = JSON.parse(fileContent).credentials;
+      } catch (error) {
+        expect(error).toBeNull();
+      }
+
+      exec(`yarn untp test -c ${storePath}`, (error, result) => {
+        if (error) {
+          console.error(`execSync error: ${error}`);
+          return;
+        }
+
+        stdout = result;
+        done();
+      });
+    });
+
+    it('should exist the volume in quantityList of objectEvent data', () => {
+      const data = fs.readFileSync(`${process.cwd()}/${credentials[0].dataPath}`, 'utf8');
+      const jsonData = JSON.parse(data);
+      expect(jsonData.quantityList[0].volume).toEqual(expect.any(String));
+    });
```

Comment:
What is the purpose of this case?

Reply:
In this case, I want to show how the final report is WARN. It warns because volume is added more than required in the schema.